### ISSUE TITLE
Add a way to get all nodes in a radius

### ISSUE THREAD

**Issue by Eh2406** [easbar/fast_paths#5]
My main use of this algorithm is in [pandana](https://github.com/UDST/pandana) ware the common need is to for each knode, calculate summary statistics about all the nodes that are within a thresholded.

**Comment by easbar** [comment]
Starting from some node p you want to find all nodes q such that the weight of the shortest path p->q is less than some threshold ?

**Comment by Eh2406** [comment]
Yes. That is what I had in mind.

**Comment by Eh2406** [comment]
Now that I think about it we could have an iterator that returns q's sorted by the shortest path p->q. If you want a k nearest: `iterator.take(k)`. If you want all in a range `iterator.take_while(|p| p->q < range)`.

If you have some pointers I may be able to give this a try.

**Comment by easbar** [comment]
Ok yes the trivial way of doing this is calculating all paths p->q, then sorting by distance and taking only those up to some limit. Do you think this should be included here (it can easily be done using the existing API) ? 

This can also be done more efficiently, but currently this is not supported. Btw in the context of routing networks this is sometimes called [isochrone calculation](https://wiki.openstreetmap.org/wiki/Isochrone).

**Comment by Eh2406** [comment]
Only when it can be done more efficiently than the alternatives. Do Contraction Hierarchies help make it more efficient than a Dijkstra's on an unprepared Graph? If so then yes it should be here. If not then I need talk to the pandana team about why they do it this way. :-P

**Comment by easbar** [comment]
Yes it can be more efficient. Even the trivial way we lined out above might be faster (on small maps at least).

> why they do it this way

You mean they are doing this already ? How would implementing it here help with Pandana ?

**Comment by Eh2406** [comment]
It is a C++ python Library, it has in the past caused our entire model to segfault. I spent today trying to diagnose a regression in performance, our model is now taking an additional 16 hours to run. As someone who doesn't know C++... let's just say I would like the rust alternatives to be as ready to be a drop-in replacement as possible.
